src/emulator.c

```diff
@@ -1326,6 +1520,12 @@ void ofono_emulator_set_indicator(struct ofono_emulator *em,
 	gboolean call;
 	gboolean callsetup;
 	gboolean waiting;
+	struct ofono_emulator *em = __ofono_atom_get_data(atom);
+
+	if (!valid_indication(em, atom, name))
```

Comment:
valid_indication() filters out indicators that are not from the currently preferred modem. The HF can only display indicators from one modem, so it would be confusing to see, say, signals from 2 different operators so the signal strength indicator flickers all the time.